### PR TITLE
README updates

### PR DESCRIPTION
**Pull request by ericvn**

### CODE REVIEW

**Comment by nmittler** [comment]
@ericvn let's hold off on this until #14076 lands (it's in the commit queue now).  It includes a major overhaul to this doc.

**Comment by ericvn** [comment]
@nmittler I'm fine with that. Just wanted to make sure I captured some things from my retrying the integration framework again over the last 2 days.

File: tests/integration/README.md [hunk 1]
```diff
@@ -29,18 +30,31 @@ case, just typing ```go test ./...``` should be sufficient to run tests.
...
+#### Notes
+
+* Running on a Mac
+  * Currently some _native_ tests fail when being run on a Mac with an error like:
+    ```
+    unable to locate an Envoy binary
+    ```
+    This is documented in this [PR](https://github.com/istio/istio/issues/13677). Once the Envoy binary is available for the Mac,
+    these tests will hopefully succeed.
+  * If one uses Docker for Mac for the kubernetes environment be sure to specify the `-istio.test.kube.minikube` parameter. The solves an error like:
```
Comment (howardjohn) [inline tests/integration/README.md]: not just d4m, anything that doesn't support LoadBalancer type

**Comment by howardjohn** [comment]
something went wrong here you have a ton of commits

**Comment by ericvn** [comment]
Sorry about that. Should have been a simple rebase. I will straighten it out.

File: tests/integration/README.md (hunk 1 above)
Comment (ericvn) [reply tests/integration/README.md]: Added comment to the command line flag as it isn't necessarily Mac specific.

**Comment by ericvn** [comment]
Split main commit into two to seperate formatting changes from actual REAME changes to allow better reviewing of commits in this PR.

File: tests/integration/README.md [hunk 2]
```diff
@@ -591,52 +586,71 @@ The test framework supports the following command-line flags:
...
         Do not cleanup resources after test completion
-                
+
   -istio.test.select string
         Comma separatated list of labels for selecting tests to run (e.g. 'foo,+bar-baz').
 
   -istio.test.hub string
-        Container registry hub to use (default "gcr.io/oztest-mixer")
-        
+        Container registry hub to use (default HUB environemnt variable)
```
Comment (nmittler) [inline tests/integration/README.md]: sp: "environment"

File: tests/integration/README.md [hunk 3]
```diff
@@ -591,52 +586,71 @@ The test framework supports the following command-line flags:
...
+
   -istio.test.tag string
-        Common Container tag to use when deploying container images (default "ozevren")
-                        
+        Common Container tag to use when deploying container images (default TAG environemnt variable)
```
Comment (nmittler) [inline tests/integration/README.md]: same comment here

File: tests/integration/README.md (hunk 2 above)
Comment (ericvn) [reply tests/integration/README.md]: Updated

File: tests/integration/README.md (hunk 3 above)
Comment (ericvn) [reply tests/integration/README.md]: Updated.

**Comment by ericvn** [comment]
Do a rebase to get pas the e2e-simpleTests-master failure?

**Pull request merged**